### PR TITLE
Ecommerce|Akash Gupta|using controller,service and dao

### PR DESCRIPTION
**Pull request by Akashhhhh**

### CODE REVIEW

File: src/main/java/ecommerce/controller/CustomerController.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    /**
+     * This method is used for providing interface
+     *
+     * @param lru cache
+     * @param con connection
+     * @throws InvalidInputException       for throwing user error
+     * @throws ApplicationRuntimeException for throwing application error
+     */
+    public void customer(LruCacheService lru, Connection con) {
```
Comment (nikhil-rss-zs) [inline src/main/java/ecommerce/controller/CustomerController.java]: avoid repeating code

File: src/main/java/ecommerce/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+     *
+     * @param custId customer id
+     * @param name   name of product
+     * @param con    connection
+     * @throws ApplicationRuntimeException for throwing application error
+     */
+    public void placeOrder(UUID custId, String name, Connection con) {
+        Map<UUID, Product> menu = null;
+        try {
+            menu = orderDao.menu(con);
```
Comment (nikhil-rss-zs) [inline src/main/java/ecommerce/controller/OrderController.java]: product service.getAllProdcuts

**Review by nikhil-rss-zs — CHANGES_REQUESTED** [review]

File: src/main/java/ecommerce/EcomApp.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        con = Connection.create();
+        lru = new LruCacheService();
+    }
+
+    /**
+     * This method is the entry point of our application
+     *
+     * @param args Unused
+     */
+    public static void main(String[] args) throws InvalidInputException, ApplicationRuntimeException {
```
Comment (nikhil-rss-zs) [inline src/main/java/ecommerce/EcomApp.java]: Why is main method throwing Exceptions ?

File: src/main/java/ecommerce/controller/CustomerController.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+        logger.info("Enter Address: ");
+        String address = sc.next();
+
+        logger.info("Enter Date of Birth: ");
+        String dob = sc.next();
+
+        Customer obj = new Customer(fName, lName, mob, email, address, dob);
+
+            Set<ConstraintViolation<Customer>> constraintViolations = validator.validate(obj);
```
Comment (nikhil-rss-zs) [inline src/main/java/ecommerce/controller/CustomerController.java]: Validations to be done in the validator class

File: src/main/java/ecommerce/controller/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+/**
+ * This class is used for managing place,delete and show order operation
+ *
+ * @author Akash gupta
+ */
+public class OrderController {
+    Scanner sc = new Scanner(System.in);
+    private static Logger logger = java.util.logging.Logger.getLogger(OrderController.class.getName());
+    OrderDao orderDao = new OrderDao();
```
Comment (nikhil-rss-zs) [inline src/main/java/ecommerce/controller/OrderController.java]: Why is orderDao initialized ?

File: src/main/java/ecommerce/controller/OrderController.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    /**
+     * This class is used for placing orders
+     *
+     * @param custId customer id
+     * @param name   name of product
+     * @param con    connection
+     */
+    public void placeOrder(UUID custId, String name, Connection con) throws ApplicationRuntimeException {
+        Map<UUID, Product> menu = null;
+
```
Comment (nikhil-rss-zs) [inline src/main/java/ecommerce/controller/OrderController.java]: Menu should be fetched from product service as this a product service functionality

File: src/main/java/ecommerce/exception/InvalidInputException.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+package ecommerce.exception;
+
+public class InvalidInputException extends Exception {
+    int erroCode;
+    String errorDesc;
+
+    public InvalidInputException(int errorCode, String errorMessage) {
```
Comment (nikhilrss) [inline src/main/java/ecommerce/exception/InvalidInputException.java:7]: initialize super(e) as well.

File: src/main/java/ecommerce/service/CustomerService.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+     *
+     * @param email   email id of customer
+     * @param address address that is updated
+     * @param lru     cache
+     * @param con     connection
+     * @throws InvalidInputException       for throwing user error
+     * @throws ApplicationRuntimeException for throwing application error
+     */
+    public void updateCustomer(String email, String address, LruCacheService lru, Connection con) throws InvalidInputException, ApplicationRuntimeException {
+        if (!validator.validateEmailId(email)) {
```
Comment (nikhil-rss-zs) [inline src/main/java/ecommerce/service/CustomerService.java]: Keep validation return methods consistent

**Review by nikhil-rss-zs — CHANGES_REQUESTED** [review]

**Review by nikhil-rss-zs — APPROVED** [review]

**Pull request merged**